agent/consul/catalog_endpoint.go

```diff
@@ -83,7 +83,7 @@ func (c *Catalog) Register(args *structs.RegisterRequest, reply *struct{}) error
 
 		// Proxies must have write permission on their destination
 		if args.Service.Kind == structs.ServiceKindConnectProxy {
-			if rule != nil && !rule.ServiceWrite(args.Service.ProxyDestination, nil) {
+			if rule != nil && !rule.ServiceWrite(args.Service.Proxy.DestinationServiceName, nil) {
```

Comment:
Is it possible for `args.Service.Proxy` to be `nil` here?

Reply:
Nope. `args.Service` is already nil checked on line 49 and `NodeService.Proxy` is not a pointer. _most_ of our internal structs embed rather than use pointers for this reason but worth checking!